Framework/Geometry/src/Objects/MeshObject.cpp

```diff
@@ -473,6 +473,18 @@ boost::shared_ptr<GeometryHandler> MeshObject::getGeometryHandler() const {
   return m_handler;
 }
 
+void MeshObject::rotate(const Kernel::Matrix<double> &rotationMatrix) {
```

Comment:
https://doxygen.mantidproject.org/nightly/d7/dca/classMantid_1_1Geometry_1_1Component.html
if this is what you mean by component, i don't see how it links to meshObject? if you mean the environment itself, thats because separate pieces of the environment may have different rotations

Reply:
Sorry I meant IObjComponent. See `ObjComponent.cpp` for implementations